astropy/coordinates/builtin_frames/heliocentric.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import (absolute_import, unicode_literals, division,
+                        print_function)
+
+from ..representation import SphericalRepresentation
+from ..baseframe import (BaseCoordinateFrame, RepresentationMapping,
+                         TimeFrameAttribute)
+from .utils import DEFAULT_OBSTIME
+
+class Heliocentric(BaseCoordinateFrame):
```

Comment:
Personally, I think not. ICRS is explicitly defined to have it's origin at the Barycenter, so ```HeliocentricICRS``` is not possible.